stan/math/prim/mat/fun/mdivide_right_tri.hpp

```diff
@@ -38,10 +38,11 @@ namespace stan {
       } else if (TriView == Eigen::Upper) {
         return transpose(mdivide_left_tri<Eigen::Lower>(transpose(A),
                                                         transpose(b)));
-      } else {
-        throw std::domain_error("triangular view must be Eigen::Lower or "
-                                "Eigen::Upper");
       }
+
+      domain_error("mdivide_left_tri",
+                   "triangular view must be Eigen::Lower or Eigen::Upper",
+                   "", "");
```

Comment:
Check errors and return from there.  We don't need
the old C style of single return points because we
use C++ RAII memory management and exceptions (those
were chokepoints that made it easier to have a single
exit point from a function that ensured all the invariants
were satisified on exit, such as not leaking memory).

> On Aug 6, 2016, at 5:15 PM, Krzysztof Sakrejda <notifications@github.com> wrote:
> 
> In stan/math/prim/mat/fun/mdivide_right_tri.hpp:
> 
> >        }
> > +
> > +      domain_error("mdivide_left_tri",
> > +                   "triangular view must be Eigen::Lower or Eigen::Upper",
> > +                   "", "");
> 
> Do we go with multiple returns and fall-through for errors in general? It's always a confusing style to me but I guess it's common (?)
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>